### DISCUSSION TITLE
Vulkan Video decoding working with .mp4 but not .mkv

### DISCUSSION THREAD

**Discussion by Brisse89** [mpv-player/mpv#15297] [General]
Am I the only one encountering this? Can't find anything related when searching around here among issues and discussions, nor anywhere else on the internet. 

I've had a bunch of problems with Vulkan Video decoding and it's been tricky to understand which ones are related and which are separate issues, but I'm no longer having kernel panics and things are narrowing down into a couple of last issues ([this](https://gitlab.freedesktop.org/mesa/mesa/-/issues/12132) being the other, which is probably not mpv's issue).

So apart from the issue I reported to mesa above, I'm having this strange issue where .mp4 files work, but .mkv doesn't. It doesn't seem to have anything to do with the codec. I find it quite strange that the container format matters in this case but it became very clear after trying a dozen of different files. The codec doesn't seem to matter. I'm thinking about opening an issue about this but I thought I would ask here first if anyone else have run into this? Whenever I play an .mkv file and try to use Vulkan Video I get no picture and there's [this](https://gist.github.com/Brisse89/97eee36e3bbad91b318121f9cdd6cf29) spam in the terminal.

Debian testing/unstable
mpv 0.38 (also tried the .39 flatpak)
ffmpeg 7:7.1-3
mesa 24.2.6-1
Linux 6.11.5-amd64
GPU: [AMD/ATI] Fiji [Radeon R9 FURY / NANO Series] [1002:7300] (rev cb)

**Comment by llyyr** [comment]
The "spam" means mpv/ffmpeg caused a gpu reset, which is a bad thing that shouldn't happen. Did you test with multiple mkv files or just this one? It's possible there's something specific about that file that can cause it. Either way open an issue if you want to report a bug

**Comment by Brisse89** [comment]
Yes, probably about 10-20 different .mkv files from various sources. 0% success rate.

Tried similar amount of .mp4 files. 100% success rate (disregarding the other issues which seem graphics driver related).

**Comment by Brisse89** [comment]
ffmpeg does not seem to have this issue. The following command works equally well whether the input is .mp4 or .mkv. I guess that raises the likelihood that this is an mpv problem.

`RADV_PERFTEST=video_decode ffmpeg -hwaccel vulkan -init_hw_device vulkan=vk:0,debug=0 -hwaccel_output_format vulkan -i <INPUT_FILE> -vf 'hwdownload,format=nv12' -pix_fmt yuv420p output.y4m`
